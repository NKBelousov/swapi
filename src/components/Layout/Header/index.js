import PropTypes from "prop-types";
import React, { memo } from "react";
import styled from "styled-components";

const Title = styled.h1`
  border-bottom: 1px solid ${props => props.theme.secondary};
  color: ${props => props.theme.secondary};
  display: block;
  font-weight: bold;
  margin: 0 auto;
  padding: 1rem 0rem;
`;

const Header = memo(({ tooltip, children }) => (
  <Title title={tooltip}>{children}</Title>
));

Header.propTypes = {
  tooltip: PropTypes.string,
  children: PropTypes.node,
};

Header.defaultProps = {
  tooltip: "",
  children: null,
};

export default Header;
