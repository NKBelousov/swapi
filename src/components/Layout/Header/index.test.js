import React from "react";
import renderer from "react-test-renderer";

import Header from "./index.js";

test("Should render correctly", () => {
  const tree = renderer
    .create(
      <Header tooltip="Test">
        <h1>Children</h1>
      </Header>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
