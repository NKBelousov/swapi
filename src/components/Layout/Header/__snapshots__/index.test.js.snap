// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<h1
  className="sc-bdVaJa gOEIiC"
  title="Test"
>
  <h1>
    Children
  </h1>
</h1>
`;
